File: project/lib/core/action/YourMoveAction.js
utils = require("../utils");
Action = require("./Action")
generators = require("../generators");


class YourMoveAction extends Action{
    constructor(unit, moves, game){
        super();

        if(unit === undefined){
            throw new utils.Exception("Not defined");
        }

        if(moves === undefined){
            throw new utils.Exception("Not defined");
        }

        this.unit = unit;


        for(var i = 0; i<game.units.length; i++){
            var u = game.units[i];
            if(!u.isBuilding()) {
                u.speed = game.units[i].getSpeed();
            }
            else{
                u.speed = 0;
            }
            if(game.hasTower(u.position)){
                u.currentHeight = utils.GAME_PARAMS.TOWER_HEIGHT;
            }
            else{
                u.currentHeight = 0;
            }
            u.sightRadius = utils.GAME_PARAMS.SIGHT_RADIUS;

            u.updateIncome(game);
        }


        var treasury = null;
        var color = game.getCurrentUnit().color;

        if(color === utils.WHITE){
            treasury = game.whiteTreasury;
        }
        else{
            treasury = game.blackTreasury;
        }

        this.foodAmount = treasury.food;
        this.stoneAmount = treasury.stone;
        this.ironAmount = treasury.iron;
        this.goldAmount = treasury.gold;

        this.foodIncome = game.getFoodIncome(color);
        this.goldIncome = game.getGoldIncome(color);
        this.stoneIncome = game.getStoneIncome(color);
        this.ironIncome = game.getIronIncome(color);

        this.width = game.map.width;
        this.height = game.map.height;

        this.units = game.units;
        //this.units = [];

        this.moves = moves;
        this.type = "Your move";

        this.obstacles = generators.makeObstacles(game);


        /*
        this.field1 = utils.create2DArray(game.map.width, game.map.height, 0);
        var sight = generators.generateSightPoints(unit, game);
        for(var i = 0; i<sight.length; i++){
            this.field1[sight[i].x][sight[i].y] = 100;
        }
        */

        //this.field1 = game.AIData.controlField;
    }
}

module.exports = YourMoveAction;

File: UnityClient/Assets/src/lib/action/YourMoveAction.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

using Assets.src.lib.entities;
using Assets.src.lib.move;


namespace Assets.src.lib.action
{
    public class YourMoveAction : Action
    {
        public List<Move> moves { get; set; }
        public Unit unit { get; set; }

        public List<Unit> units { get; set; }

        public int foodIncome { get; set; }
        public int goldIncome { get; set; }
        public int ironIncome { get; set; }
        public int stoneIncome { get; set; }

        public int foodAmount { get; set; }
        public int goldAmount { get; set; }
        public int ironAmount { get; set; }
        public int stoneAmount { get; set; }

        public List<Obstacle> obstacles { get; set; }
        public int width;
        public int height;
    }
}
